internal/langserver/handlers/did_open.go

```diff
@@ -38,8 +37,6 @@ func (svc *service) TextDocumentDidOpen(ctx context.Context, params lsp.DidOpenT
 		return err
 	}
 
-	ctx = lsctx.WithLanguageId(ctx, params.TextDocument.LanguageID)
```

Comment:
I don't know if there's actually any other method that has the same structure that will benefit from this but given that we already do some unmarshaling prior to passing the request to the handler I assume we may as well unmarshal the LanguageID there instead of here.